### PR TITLE
Add polymorphic empty->nil function

### PR DESCRIPTION
**Pull request by coconutpalm**
- \+ moved any? predicate to be next to the other predicates.

See the doc string for details.

### CODE REVIEW

File: src/clj_foundation/patterns.clj [hunk 1]
```diff
@@ -137,6 +132,12 @@
     value))
 
 
+(defn any?
```
Comment (baguthrie) [inline src/clj_foundation/patterns.clj:135]: Why not `clojure.core/some`?

Comment (baguthrie) [inline src/clj_foundation/patterns.clj:135]: I know this isn't a serious change from here, but thought I'd mention.

**Comment by baguthrie** [comment]
👍

**Comment by recursive-automata** [comment]
👍

**Pull request merged**